test/Spec/Integration.hs

```diff
@@ -65,143 +64,142 @@ import Test.Plutip.Predicate (
-          "Fails because outside validity interval"
-          (initAda [100])
-          (withContract $ const failingTimeContract)
-          [shouldFail]
-      , assertExecution
-          "Passes validation with exact time range checks"
-          (initAda [100])
-          (withContract $ const successTimeContract)
-          [shouldSucceed]
-      , -- always fail validation test
-        let errCheck e = "I always fail" `isInfixOf` pack (show e)
-         in assertExecution
-              "Always fails to validate"
-              (initAda [100])
-              (withContract $ const lockThenFailToSpend)
-              [ shouldFail
-              , errorSatisfies "Fail validation with 'I always fail'" errCheck
-              ]
-      ]
+            withContractAs 1 $ -- run contract which result will be checked
+              \[pkh1] -> payTo pkh1 10_000_000
+        )
+        [shouldSucceed]
+    , -- Tests with assertions on Contract return value
+      assertExecution
+        "Initiate wallet and get UTxOs"
+        (initAda [100])
+        (withContract $ const getUtxos)
+        [ yieldSatisfies "Returns single UTxO" ((== 1) . Map.size)
+        ]
+    , let initFunds = 10_000_000
+       in assertExecution
+            "Should yield own initial Ada"
+            (initLovelace [toEnum initFunds])
+            (withContract $ const ownValue)
+            [ shouldYield (lovelaceValueOf $ toEnum initFunds)
+            ]
+    , -- Tests with assertions on state
+      let initFunds = 10_000_000
+       in assertExecution
+            "Puts own UTxOs Value to state"
+            (initLovelace [toEnum initFunds])
+            (withContract $ const ownValueToState)
+            [ stateIs [lovelaceValueOf $ toEnum initFunds]
+            , Predicate.not $ stateSatisfies "length > 1" ((> 1) . length)
+            ]
+    , -- Tests with assertions on failure
+      let expectedErr = ConstraintResolutionContractError OwnPubKeyMissing
+          isResolutionError = \case
+            ConstraintResolutionContractError _ -> True
+            _ -> False
+       in assertExecution
+            ("Contract which throws `" <> show expectedErr <> "`")
+            (initAda [100])
+            (withContract $ const getUtxosThrowsErr)
+            [ shouldThrow expectedErr
+            , errorSatisfies "Throws resolution error" isResolutionError
+            , Predicate.not $ failReasonSatisfies "Throws exception" isException
+            ]
+    , let checkException = \case
+            CaughtException e -> isJust @ErrorCall (fromException e)
+            _ -> False
+       in assertExecution
+            "Contract which throws exception"
+            (initAda [100])
+            (withContract $ const getUtxosThrowsEx)
+            [ shouldFail
+            , Predicate.not shouldSucceed
+            , failReasonSatisfies "Throws ErrorCall" checkException
+            ]
+    , -- tests with assertions on execution budget
+      assertExecution
```

Comment:
Strange that this test didn't fail in CI. W/o `BudgetCounting` option there should not be budget info in execution result, iirc budget related predicates should fail if there is no budget info.

Reply:
Predicates fail with no budget info, but currently  we set [pcCollectStats to True](https://github.com/mlabs-haskell/plutip/blob/ab8f2e8b01724147d28b93c93113ecd925c5e812/src/Test/Plutip/Internal/BotPlutusInterface/Run.hs#L112) unconditionaly, so budget info and logs are always in present `ExecutionResult`.

To me this looks fine, budget and tracing options decide printing only. I've checked if collecting these influences performance, with a timer xd, and seemingly it doesn't. But if we want pcCollect[..] options decided based on user passed options to `assertExecutionWith` I can add that, this information could be placed in `ContractEnvironment`.

Reply:
I see. Lets live it like that, I guess. We can change that if it will become an issue.

>end enabling assertions for them

should be removed from [here](https://github.com/mlabs-haskell/plutip/pull/80#discussion_r892248171) then.

Can you also change couple tests to use `assertExecutionWith`, as we provide `Integration.hs` as an example for users too.

Reply:
okey did, and updated that old readme, but only by rewriting the cases as they are to use `assertExecution`